In [23]:
import pandas as pd
import numpy as np

In [50]:
df = pd.read_excel("/home/jupyter/src/dataviz/podcastcount/2LMarketBasket.xlsx")

In [51]:
df.shape

(10931, 44)

In [52]:
df.rename(columns = {'2L+ TKI Target':'2Lplus',
                     'LTD Reach - Live':'LiveRepTouch',
                     'LTD Reach - All Reach Methods':'AnyTouch',
                     'April Marketing Campaign - Email Open Count':'AprilEmailOpens'},inplace = True)

In [53]:
list(df)

['NPI',
 'HCP ID',
 'First Name',
 'Last Name',
 'Specialty',
 'Sales Tier',
 'Facility Tier',
 '2Lplus',
 'AnyTouch',
 'LiveRepTouch',
 'AprilEmailOpens',
 'Total GIST Patients',
 'Total Treated GIST Patients',
 'Total 3L+ GIST Patients',
 'Total Ayva GIST Patients',
 'Komodo Clinical Experience Score',
 'Komodo Clinical Volume Score',
 'GLEEVEC Units - 52 Week',
 'GLEEVEC Units - 26 Week',
 'Sutent Units - 52 Week',
 'Sutent Units - 26 Week',
 'Stivarga Units - 52 Week',
 'Stivarga Units - 26 Week',
 'Ayva Units - 52 Week',
 'Ayva Units - 26 Week',
 'Other TKI Units - 52 Week',
 'Other TKI Units - 26 Week',
 '2L+ TKI Units - 52 Week',
 '2L+ TKI Units - 26 Week',
 'Field Reported 2L+',
 'LTD QINLOCK Rx (HCP)',
 'Region',
 'Territory',
 'Facility ID',
 'Facility Name',
 'Facility Address',
 'Facility Address 2',
 'Facility City',
 'Facility State',
 'Facility Zip',
 'Facility Phone',
 'Facility Fax',
 'Network/Parent',
 'Facility Segment']

In [54]:
pd.pivot_table(df,index=["2Lplus","Region"])

2L+ TKI Units - 26 Week  2L+ TKI Units - 52 Week  \
2Lplus Region                                                          
No     Central                     0.132107                 0.244147   
       Great Lakes                 0.209841                 0.399421   
       Northeast                   0.023739                 0.051929   
       South                       0.161572                 0.269287   
       Southeast                   0.082982                 0.205345   
       West                        0.219638                 0.400517   
Yes    Central                     2.041885                 3.921466   
       Great Lakes                 2.004902                 4.004902   
       Northeast                   2.657754                 4.791444   
       South                       1.932990                 3.886598   
       Southeast                   1.862559                 3.530806   
       West                        1.322581                 2.562724   

                    AnyTouch  AprilEmailOpens  Ayva Units - 26 Week  \
2Lplus Region                                                         
No     Central      1.774658         3.540000              0.005017   
       Great Lakes  2.514036         3.204724              0.000000   
       Northeast    1.439294         3.120846              0.000000   
       South        2.276278         3.047210              0.000000   
       Southeast    3.586261         3.040590              0.000000   
       West         1.839329         3.388889              0.002584   
Yes    Central      5.138393         3.181818              0.026178   
       Great Lakes  6.476596         3.638298              0.014706   
       Northeast    4.676190         2.808511              0.000000   
       South        6.131455         3.409091              0.041237   
       Southeast    8.902834         2.756098              0.052133   
       West         4.927978         3.016129              0.050179   

                    Ayva Units - 52 Week  Facility Zip  Field Reported 2L+  \
2Lplus Region                                                                
No     Central                  0.005017  64390.398630            0.000000   
       Great Lakes              0.000000  34963.549595            0.000000   
       Northeast                0.000000  10198.255519            0.000000   
       South                    0.000000  70357.611506            0.000000   
       Southeast                0.000000  31826.834819            0.000000   
       West                     0.002584  92779.757794            0.000000   
Yes    Central                  0.047120  60466.223214            1.489247   
       Great Lakes              0.014706  34494.000000            1.656805   
       Northeast                0.016043   9371.661905            1.333333   
       South                    0.103093  71173.211268            1.356688   
       Southeast                0.061611  31603.441296            1.241935   
       West                     0.064516  93192.650970            1.484746   

                    GLEEVEC Units - 26 Week  GLEEVEC Units - 52 Week  ...  \
2Lplus Region                                                         ...   
No     Central                     2.070234                 4.133779  ...   
       Great Lakes                 2.043415                 4.046310  ...   
       Northeast                   2.243323                 4.379822  ...   
       South                       2.314410                 4.499272  ...   
       Southeast                   2.032349                 3.893108  ...   
       West                        2.240310                 4.447028  ...   
Yes    Central                     4.062827                 8.272251  ...   
       Great Lakes                 5.171569                10.421569  ...   
       Northeast                   6.759358                13.572193  ...   
       South                       4.577320                 9.195876  ...   

In [55]:
pd.pivot_table(df,index=["2Lplus","Region"],values=["AprilEmailOpens"],aggfunc=[np.sum])

sum
                   AprilEmailOpens
2Lplus Region                     
No     Central               885.0
       Great Lakes           814.0
       Northeast            1033.0
       South                 710.0
       Southeast             824.0
       West                  671.0
Yes    Central               140.0
       Great Lakes           171.0
       Northeast             132.0
       South                 150.0
       Southeast             113.0
       West                  187.0

In [57]:
pd.pivot_table(df,index=["2Lplus","Region"],values=["Total 3L+ GIST Patients","AprilEmailOpens"],aggfunc=[np.sum])

sum                        
                   AprilEmailOpens Total 3L+ GIST Patients
2Lplus Region                                             
No     Central               885.0                     6.0
       Great Lakes           814.0                    10.0
       Northeast            1033.0                     1.0
       South                 710.0                     7.0
       Southeast             824.0                     9.0
       West                  671.0                     8.0
Yes    Central               140.0                    31.0
       Great Lakes           171.0                    44.0
       Northeast             132.0                    53.0
       South                 150.0                    45.0
       Southeast             113.0                    49.0
       West                  187.0                    34.0

In [32]:
pd.pivot_table(df,index=["2Lplus","Region"],
               values=["Total 3L+ GIST Patients","Total Treated GIST Patients"],
               aggfunc=[np.sum])

sum                            
                   Total 3L+ GIST Patients Total Treated GIST Patients
2Lplus Region                                                         
No     Central                         6.0                       418.0
       Great Lakes                    10.0                       499.0
       Northeast                       1.0                       502.0
       South                           7.0                       530.0
       Southeast                       9.0                       488.0
       West                            8.0                       327.0
Yes    Central                        31.0                       382.0
       Great Lakes                    44.0                       472.0
       Northeast                      53.0                       610.0
       South                          45.0                       431.0
       Southeast                      49.0                       524.0
       West                           34.0                       522.0

In [31]:
df["2Lplus"].value_counts()

No     9441
Yes    1490
Name: 2Lplus, dtype: int64

+ (41) 3L Patients among 9441 HCPs **THAT ARE NOT** 2L plus
+ (256) 3L Patients among 1490 HCPs **THAT ARE** 2L plus

In [33]:
pd.pivot_table(df,index=["2Lplus"],
               values=["Total 3L+ GIST Patients"],
               columns=["Region"],
               aggfunc=[np.sum])

sum                                            
       Total 3L+ GIST Patients                                            
Region                 Central Great Lakes Northeast South Southeast  West
2Lplus                                                                    
No                         6.0        10.0       1.0   7.0       9.0   8.0
Yes                       31.0        44.0      53.0  45.0      49.0  34.0

In [42]:
table = pd.pivot_table(df,index=["2Lplus","Region"],
               values=["Total 3L+ GIST Patients","Total Treated GIST Patients","LiveRepTouch","AnyTouch"],
               aggfunc=[np.sum])
table

sum                                       \
                   AnyTouch LiveRepTouch Total 3L+ GIST Patients   
2Lplus Region                                                      
No     Central         2591          652                     6.0   
       Great Lakes     4030          801                    10.0   
       Northeast       2608          234                     1.0   
       South           3205          492                     7.0   
       Southeast       6839         1159                     9.0   
       West            2301          544                     8.0   
Yes    Central         1151          408                    31.0   
       Great Lakes     1522          340                    44.0   
       Northeast        982          159                    53.0   
       South           1306          262                    45.0   
       Southeast       2199          550                    49.0   
       West            1779          477                    34.0   

                                                
                   Total Treated GIST Patients  
2Lplus Region                                   
No     Central                           418.0  
       Great Lakes                       499.0  
       Northeast                         502.0  
       South                             530.0  
       Southeast                         488.0  
       West                              327.0  
Yes    Central                           382.0  
       Great Lakes                       472.0  
       Northeast                         610.0  
       South                             431.0  
       Southeast                         524.0  
       West                              522.0

In [68]:
table = pd.pivot_table(df,index=["2Lplus","Region"],
               values=["Total 3L+ GIST Patients","LiveRepTouch","AnyTouch","AprilEmailOpens"],
               aggfunc=[np.sum])
table

sum                               \
                   AnyTouch AprilEmailOpens LiveRepTouch   
2Lplus Region                                              
No     Central         2591           885.0          652   
       Great Lakes     4030           814.0          801   
       Northeast       2608          1033.0          234   
       South           3205           710.0          492   
       Southeast       6839           824.0         1159   
       West            2301           671.0          544   
Yes    Central         1151           140.0          408   
       Great Lakes     1522           171.0          340   
       Northeast        982           132.0          159   
       South           1306           150.0          262   
       Southeast       2199           113.0          550   
       West            1779           187.0          477   

                                            
                   Total 3L+ GIST Patients  
2Lplus Region                               
No     Central                         6.0  
       Great Lakes                    10.0  
       Northeast                       1.0  
       South                           7.0  
       Southeast                       9.0  
       West                            8.0  
Yes    Central                        31.0  
       Great Lakes                    44.0  
       Northeast                      53.0  
       South                          45.0  
       Southeast                      49.0  
       West                           34.0

In [72]:
list(table)

[('sum', 'AnyTouch'),
 ('sum', 'AprilEmailOpens'),
 ('sum', 'LiveRepTouch'),
 ('sum', 'Total 3L+ GIST Patients'),
 ('EmailOpenPerTouch', '')]

In [70]:
table

sum                               \
                   AnyTouch AprilEmailOpens LiveRepTouch   
2Lplus Region                                              
No     Central         2591           885.0          652   
       Great Lakes     4030           814.0          801   
       Northeast       2608          1033.0          234   
       South           3205           710.0          492   
       Southeast       6839           824.0         1159   
       West            2301           671.0          544   
Yes    Central         1151           140.0          408   
       Great Lakes     1522           171.0          340   
       Northeast        982           132.0          159   
       South           1306           150.0          262   
       Southeast       2199           113.0          550   
       West            1779           187.0          477   

                                           EmailOpenPerTouch  
                   Total 3L+ GIST Patients                    
2Lplus Region                                                 
No     Central                         6.0               foo  
       Great Lakes                    10.0               foo  
       Northeast                       1.0               foo  
       South                           7.0               foo  
       Southeast                       9.0               foo  
       West                            8.0               foo  
Yes    Central                        31.0               foo  
       Great Lakes                    44.0               foo  
       Northeast                      53.0               foo  
       South                          45.0               foo  
       Southeast                      49.0               foo  
       West                           34.0               foo